src/autowiring/AnySharedPointer.h

```diff
@@ -68,6 +68,8 @@ struct AnySharedPointer {
     return obj && try_assign(obj);
   }
 
+  const std::shared_ptr<void>& as_void(void) const { return m_ptr; }
+
```

Comment:
In fact, I have something like that now, it's called [`as`](https://github.com/leapmotion/autowiring/pull/895/files?diff=unified#diff-5de4567ee35f6903d932573f2bc0c4b2R74)